src/Collection.php

```diff
@@ -94,7 +96,13 @@ public static function range($start = 0, $end = null, $step = 1)
     public function getIterator()
     {
         if ($this->inputFactory) {
-            $this->input = call_user_func($this->inputFactory);
+            $input = call_user_func($this->inputFactory);
+
+            if (is_array($input)) {
+                $input = new RecursiveArrayIterator($input);
+            }
+
```

Comment:
Right, but Scrutinizer [fails with](https://scrutinizer-ci.com/g/DusanKasan/Knapsack/inspections/113b8689-f53c-43be-b981-3ff2e6c9340f/issues/files/src/Collection.php?status=failed&conditionIds[0]=172189) 

> The return type of `return $this->input;` (`object|integer|double|string|null|boolean`) is incompatible with the return type declared by the interface `IteratorAggregate::getIterator` of type `Traversable`. 

if we don't throw.